From b6d132436e70390be2bc09b5ec8adc28431db5ee Mon Sep 17 00:00:00 2001
From: Sabin Floares <sabin.floares@intel.com>
Date: Tue, 15 Nov 2016 12:58:11 +0200
Subject: [PATCH] Fix compile list.

Mailing list URL: http://mielke.cc/pipermail/brltty/2016-October/014331.html

The service package libraries should be in the link list
instead of in the compile list.

For systemd enabled configurations SERVICE_LIBS = -lsystemd, and
$(SERVICE_LIBS) appears in PROGRAM_OBJECTS, so make will be
looking for an "-lsystemd" file.

This fix is included in brltty upstream version. This shouldn't be
needed with the next available release.

fix by Samuel Thibault.
---
 config.mk.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/config.mk.in
+++ b/config.mk.in
@@ -235,7 +235,7 @@ TUNE_OBJECTS = tune.$O notes.$O $(BEEP_OBJECTS) $(PCM_OBJECTS) $(MIDI_OBJECTS) $
 ASYNC_OBJECTS = async_handle.$O async_data.$O async_wait.$O async_alarm.$O async_task.$O async_io.$O async_event.$O async_signal.$O thread.$O
 BASE_OBJECTS = log.$O addresses.$O file.$O device.$O parse.$O variables.$O datafile.$O unicode.$O $(CHARSET_OBJECTS) timing.$O $(ASYNC_OBJECTS) queue.$O lock.$O $(DYNLD_OBJECTS) $(PORTS_OBJECTS) $(SYSTEM_OBJECTS)
 OPTIONS_OBJECTS = options.$O $(PARAMS_OBJECTS)
-PROGRAM_OBJECTS = program.$O $(PGMPATH_OBJECTS) $(SERVICE_OBJECTS) $(SERVICE_LIBS) pid.$O $(OPTIONS_OBJECTS) $(BASE_OBJECTS)
+PROGRAM_OBJECTS = program.$O $(PGMPATH_OBJECTS) $(SERVICE_OBJECTS) pid.$O $(OPTIONS_OBJECTS) $(BASE_OBJECTS)
 
 CC = @CC@
 CPPFLAGS = -I$(BLD_DIR) -I$(SRC_DIR) -I$(BLD_TOP:/=)/$(PGM_DIR) -I$(SRC_TOP:/=)/$(PGM_DIR) -I$(SRC_TOP:/=)/$(HDR_DIR) -I$(BLD_TOP:/=) -I$(SRC_TOP:/=) @CPPFLAGS@ @DEFS@
@@ -248,7 +248,7 @@ LIBCXXFLAGS = $(CXXFLAGS) @LIBCXXFLAGS@
 
 LD = @LD@
 LDFLAGS = @LDFLAGS@
-LDLIBS = $(ICU_LIBS) $(POLKIT_LIBS) $(SYSTEM_LIBS) @LIBS@
+LDLIBS = $(ICU_LIBS) $(POLKIT_LIBS) $(SYSTEM_LIBS) $(SERVICE_LIBS) @LIBS@
 
 MKOBJ = @MKOBJ@
 MKMOD = @MKMOD@
-- 
1.9.1

